### ISSUE TITLE
error in process sentinel: cide--idb-file-to-obj: Wrong type argument: hash-table-p, nil

### ISSUE THREAD

**Issue by stardiviner** [atilaneves/cmake-ide#203]
I got error on opening C++ source code file.
```
error in process sentinel: cide--idb-file-to-obj: Wrong type argument: hash-table-p, nil
error in process sentinel: Wrong type argument: hash-table-p, nil
```
And here is my config:
```elisp
(use-package cmake-ide
  :ensure t
  :defer t
  :init (cmake-ide-setup))
```

**Comment by atilaneves** [comment]
I've seen that error before. Does it always happen? Normally it's because there's an issue with the compilation database.

**Comment by stardiviner** [comment]
It reports error everytime I open .cpp file.

**Comment by atilaneves** [comment]
On a non-cmake project?

**Comment by stardiviner** [comment]
Yes. Maybe cmake-ide need to add condtion detect?

**Comment by stardiviner** [comment]
Hi, @atilaneves sorry for bothering you. Is there any update on this issue? Still seeing this error message.

**Comment by atilaneves** [comment]
Do you have a minimal repro I can use?

**Comment by stardiviner** [comment]
When I try to create an reproduce environment, I figured out why now, The CMakeLists.txt file is empty. This caused the problem. After I write it up. The problem is gone.
Should cmake-ide try to report more explicit warning about CMakeLists.txt file is empty?

**Comment by stardiviner** [comment]
Elisp can read and detect whether CMakeLists.txt is empty.
It can be done like this:
```elisp
(with-temp-buffer
  (insert-file-contents "CMakeLists.txt")
  (let ((content (buffer-substring-no-properties (point-min) (point-max))))
    (when (string-empty-p content)
      ...)))
```
This detection is very strict, only when `content` is `""` empty string. If `" "` or `TAB` etc will failed. Need more smart rules.
Don't know how to use Elisp regexp or other ways to apply.

**Comment by atilaneves** [comment]
> Elisp can read and detect whether CMakeLists.txt is empty.
> It can be done like this:
> 
> ```emacs-lisp
> (with-temp-buffer
>   (insert-file-contents "CMakeLists.txt")
>   (let ((content (buffer-substring-no-properties (point-min) (point-max))))
>     (when (string-empty-p content)
>       ...)))
> ```
> 
> This detection is very strict, only when `content` is `""` empty string. If `" "` or `TAB` etc will failed. Need more smart rules.
> Don't know how to use Elisp regexp or other ways to apply.

Right. What I meant by "obviously a completely empty CMakeLists.txt is a problem" is that this is an easy case to check, but that's a particular example that doesn't generalise. For instance, a non-empty file with a line containing "42" isn't empty, but it might as well be from cmake's perspective.

**Comment by stardiviner** [comment]
Does CMake provide some parser command or lint command? I checked `man cmake`, not found.
Then I searched "cmake parse", found one Python project. https://github.com/ijt/cmakelists_parsing
But using external tool like this cmakelists_parsing is not a good idea.
For now, seems really no good solution.
But I think cmake-ide can add empty detection at first. At least it can detect real empty file, and save place for future solution.
